R/map.R

```diff
@@ -225,6 +293,60 @@ with_indexed_errors <- function(expr, i, names = NULL, error_call = caller_env()
   )
 }
 
+with_parallel_indexed_errors <- function(expr, interrupt_expr = NULL, error_call = caller_env()) {
+  withCallingHandlers(
+    expr,
+    error = function(cnd) {
+      location <- cnd$location
+      iname <- cnd$name
+      cli::cli_abort(
+        c(
+          i = "In index: {location}.",
+          i = if (length(iname) && nzchar(iname)) "With name: {iname}."
+        ),
+        location = location,
+        name = iname,
+        parent = cnd$parent,
+        call = error_call,
+        class = "purrr_error_indexed"
+      )
+    },
+    interrupt = function(cnd) {
+      interrupt_expr
+    }
+  )
+}
+
+ensure_parallel_dependencies <- function(error_call = NULL) {
+
```

Comment:
Is the a reason to not use `rlang::check_installed()` here?

Reply:
Hmm good point, let me switch.

Reply:
Done in [304af59](https://github.com/tidyverse/purrr/pull/1163/commits/304af592b4696a2669d22cd696ea52a9b62b0aad). Saves a lot of boilerplate!

Reply:
Changed to still cache the result in [4b4066a](https://github.com/tidyverse/purrr/pull/1163/commits/4b4066ac67eb0116eb0b802f2f83abb5aa82a69c) as `rlang::check_installed()` is rather expensive.